File: bone-frontend/apps/bone-iam-app/src/utils/accountStatus.ts
/** 将后端 AccountStatus 枚举或数字统一为 0/1/2 */
export function accountStatusToCode(status: unknown): number {
  if (typeof status === 'number' && !Number.isNaN(status)) {
    return status;
  }
  if (typeof status === 'string') {
    const map: Record<string, number> = {
      DISABLED: 0,
      ENABLED: 1,
      LOCKED: 2,
    };
    return map[status] ?? 0;
  }
  return 0;
}

File: bone-platform/bone-iam/src/main/java/com/bone/iam/domain/model/account/vo/AccountStatus.java
package com.bone.iam.domain.model.account.vo;

import com.bone.core.exception.DomainException;

public enum AccountStatus {
  DISABLED(0, "禁用"),
  ENABLED(1, "启用"),
  LOCKED(2, "锁定");

  private final int code;
  private final String description;

  AccountStatus(int code, String description) {
    this.code = code;
    this.description = description;
  }

  public int getCode() {
    return code;
  }

  public String getDescription() {
    return description;
  }

  public static AccountStatus of(int code) {
    for (AccountStatus status : values()) {
      if (status.code == code) return status;
    }
    throw new DomainException("无效的状态码: " + code);
  }
}
